### PR TITLE
Remove `DDPSpawnStrategy.get_mp_spawn_kwargs` in favor of launchers

### LINKED ISSUE #10895: Deprecate or re-purpose the `get_mp_spawn_kwargs`  hook in spawn plugins
## Proposed refactor

Discuss and refactor `TPU/DDPSpawnPlugin.get_mp_spawn_kwargs`.

### Motivation

As part of our effort to converge to a stable and well organized Strategy+Accelerator API #10416  we are looking for opportunities to simplify a couple of hooks. One is the get_mp_spawn_kwargs in the Spawn-plugins. 

```py
    def get_mp_spawn_kwargs(self, trainer: Optional["pl.Trainer"] = None) -> Dict[str, Any]:
        return {
            "nprocs": len(self.parallel_devices),
            "start_method": self.start_method,
        }
```
### Pitch

**Proposal 1:** 
- Remove/deprecate the method
- Directly use `start_method` as an attribute and/or constructor argument
- `nprocs` can be set already in the constructor
- A custom implementation that needs to pass additional arguments would instead override the new `spawn()` method.

**Proposal 2:**
- Keep it
- Remove the Trainer argument
- Mark it as protected, since the method is only used internally by the Plugin

**Proposal 3:**
- Keep as is

### Additional context

Original discussion started in #10034 by @four4fish  

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @justusschock @awaelchli @akihironitta @tchaton

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #10895 
Follow up to #11643 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @justusschock @kaushikb11 @awaelchli @akihironitta @rohitgr7

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/strategies/launchers/xla_spawn.py [hunk 1]
```diff
@@ -42,8 +45,17 @@ class _XLASpawnLauncher(_SpawnLauncher):
...
         - This launcher requires all objects to be pickleable.
         - It is important that the entry point to the program/script is guarded by ``if __name__ == "__main__"``.
+
+    Args:
+        strategy: A reference to the strategy that is used together with this launcher
+        start_method: Start method for `~torch_xla.distributed.xla_multiprocessing.spawn`. Accepted options are
+            ``'spawn'`` or ``'fork'``.
     """
 
+    def __init__(self, strategy: "Strategy", start_method: str = "fork") -> None:
```
[the remark this answers is not in the document]
Comment (kaushikb11) [reply pytorch_lightning/strategies/launchers/xla_spawn.py:55]: @awaelchli Yup, we use `start_method="fork"` is minimize the host memory. Also, `fork` start method is required for XLA's `MpModelWrapper` method. We could remove it from the constructor.

**Review by kaushikb11 — APPROVED** [review]
LGTM! Just `start_method` needs to be removed from `_XLASpawnLauncher`'s constructor.

Also, loving the Launcher Strategy interface.

File: CHANGELOG.md [hunk 2]
```diff
@@ -541,7 +541,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed `FitLoop.current_epoch` getter and setter ([#11562](https://github.com/PyTorchLightning/pytorch-lightning/pull/11562))
 
 
- - Removed access to `_short_id` in `NeptuneLogger` ([#11517](https://github.com/PyTorchLightning/pytorch-lightning/pull/11517))
+- Removed access to `_short_id` in `NeptuneLogger` ([#11517](https://github.com/PyTorchLightning/pytorch-lightning/pull/11517))
+
+
+- Removed `get_mp_spawn_kwargs` from `DDPSpawnStrategy` and `TPUSpawnStrategy` in favor of configuration in the Launcher ([#11966](https://github.com/PyTorchLightning/pytorch-lightning/pull/11966))
```
Comment (kaushikb11) [inline CHANGELOG.md:547]:
```suggestion
- Removed `get_mp_spawn_kwargs` from `DDPSpawnStrategy` and `TPUSpawnStrategy` in favor of configuration in the `_SpawnLauncher` ([#11966](https://github.com/PyTorchLightning/pytorch-lightning/pull/11966))
```

File: pytorch_lightning/strategies/launchers/xla_spawn.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (awaelchli) [reply pytorch_lightning/strategies/launchers/xla_spawn.py:55]: Ok, thanks for confirming. Doesn't make to have it in the constructor then.
